x/emissions/module/rewards/scores.go

```diff
@@ -13,13 +13,40 @@ import (
  consideration to generate the score at the most recent time step.
 */
 
+type ReputersMappedLosses struct {
+	CombinedLosses         map[string]float64
+	NaiveLosses            map[string]float64
+	OneOutInfererLosses    map[string][]float64
+	OneOutForecasterLosses map[string][]float64
+	OneInForecasterLosses  map[string][]float64
+}
+
 // GenerateReputerScores calculates and persists scores for reputers based on their reported losses.
 func GenerateReputerScores(ctx sdk.Context, keeper keeper.Keeper, topicId uint64, block int64, reportedLosses types.ReputerValueBundles) ([]types.Score, error) {
-	// Get reputers data
+	// 1) Fetch all workers for the topic to determine expected report coverage
+	allWorkers, err := keeper.GetTopicWorkers(ctx, topicId)
```

Comment:
Cuz I need to know all registered ones to be able to know which ones are missing

Reply:
You really only need to know only those that are in any of the reputer's reports though right? doesn't matter if a worker is registered for a topic but not included in any report of any reputer (perhaps because they didn't actually submit an inference), right?

Reply:
So what I'd do here is make a superset of all the workers considered by all reputers' reports. There should be 2 such lists: 1 for inferers, 1 for forecasters. Then, for each "class" of loss `L_i, L_ij, L_ik, L^-_i, L^-_ij, L^-_ik, L^+_ik`, you impute `NaN` if the reputer lacks a value for the inferer or forecaster.

Note that this implies multiple changes here and below:
1. Don't loop over all registered workers, as that is unbounded.
2. You need 2 arrays (for inferers, forecasters), not just 1 for all workers.
3. You need to use each of those^ arrays below in different sports depending on the "class" of loss being looped over.